pkg/runtime/framework/plugins/torch/torch.go

```diff
@@ -140,3 +137,31 @@ func (t *Torch) EnforceMLPolicy(info *runtime.Info, trainJob *trainer.TrainJob)
 
 	return nil
 }
+
+func (t *Torch) Validate(runtimeJobTemplate client.Object, runtimeInfo *runtime.Info, oldObj, newObj *trainer.TrainJob) (admission.Warnings, field.ErrorList) {
+	var allErrs field.ErrorList
+	specPath := field.NewPath("spec")
+
+	if newObj.Spec.Trainer != nil {
+		numProcPerNodePath := specPath.Child("trainer").Child("numProcPerNode")
+		if runtimeInfo.RuntimePolicy.MLPolicy != nil &&
+			runtimeInfo.RuntimePolicy.MLPolicy.Torch != nil && newObj.Spec.Trainer.NumProcPerNode != nil {
+			numProcPerNode := *newObj.Spec.Trainer.NumProcPerNode
+			if numProcPerNode.Type == intstr.String {
+				allowedStringValList := []string{"auto", "cpu", "gpu"}
+				if !slices.Contains(allowedStringValList, numProcPerNode.StrVal) {
+					allErrs = append(allErrs, field.Invalid(numProcPerNodePath, newObj.Spec.Trainer.NumProcPerNode, "should have an int value or auto/cpu/gpu"))
+				}
+			}
+		}
+
+		if slices.ContainsFunc(newObj.Spec.Trainer.Env, func(x corev1.EnvVar) bool {
+			return strings.HasPrefix(x.Name, constants.TorchEnvNamePrefix)
+		}) {
```

Comment:
```suggestion
		torchEnvs := sets.New[string]()
		for _, env := range newObj.Spec.Trainer.Env {
			if strings.HasPrefix(env.Name) {
				torchEnvs.Insert(env.Name)
			}
		}
```

Could we collect all torch envs so that we can notify users all invalid environment variables?